bot/cogs/queue.py

```diff
@@ -178,78 +161,65 @@ async def join(self, ctx):
             title = f'Unable to add **{ctx.author.display_name}**: Their account is not linked'
         else:  # Message author is linked
             player = await self.bot.api_helper.get_player(ctx.author)
+            await self.bot.db_helper.insert_users(ctx.author)
+            queue_ids = await self.bot.db_helper.get_queued_users(ctx.guild)
+            queue = [self.bot.get_user(user_id) for user_id in queue_ids]
+            guild_data = await self.bot.db_helper.get_guild(ctx.guild)
+            capacity = guild_data['capacity']
 
-            if ctx.author in queue.active:  # Author already in queue
+            if ctx.author in queue:  # Author already in queue
                 title = f'**{ctx.author.display_name}** is already in the queue'
-            elif len(queue.active) >= queue.capacity:  # Queue full
+            elif len(queue) >= capacity:  # Queue full
                 title = f'Unable to add **{ctx.author.display_name}**: Queue is full'
             elif not player:  # ApiHelper couldn't get player
                 title = f'Unable to add **{ctx.author.display_name}**: Cannot verify match status'
             elif player.in_match:  # User is already in a match
                 title = f'Unable to add **{ctx.author.display_name}**: They are already in a match'
             else:  # User can be added
-                queue.active.append(ctx.author)
+                await self.bot.db_helper.insert_queued_users(ctx.guild, ctx.author)
+                queue += [ctx.author]
                 title = f'**{ctx.author.display_name}** has been added to the queue'
 
                 # Check and burst queue if full
-                if len(queue.active) == queue.capacity:
+                if len(queue) == capacity:
                     try:
-                        all_readied = await self.start_match(ctx, queue.active)
+                        all_readied = await self.start_match(ctx, queue)
```

Comment:
I feel like this match starting logic doesn't really belong here since it's just handling the user joining the queue. It does the job for now but it'd be worth looking at moving this in the future.

Reply:
Yeah I was thinking the same thing. My idea was to have a cog for managing the whole match starting process. Could maybe move the team draft and map draft stuff in there too.

Reply:
Yeah it'd be good to move them into their own respective classes. When doing this we could perhaps look at a way to allow a user to write their own custom team drafting / map drafting logic via a plugin or something along those lines.

Reply:
Yeah I am storing it on the guild right now because the guilds still only have one queue each.